src/Microsoft.AspNet.Razor/CodeGenerators/CSharpTagHelperCodeRenderer.cs

```diff
@@ -229,15 +229,9 @@ protected virtual string GenerateUniqueId()
                     descriptor.Attributes.Any(attributeDescriptor => attributeDescriptor.IsNameMatch(attributeName)));
 
 
-                // Bound attributes have associated descriptors.
-                if (associatedDescriptors.Any())
+                // Bound attributes have associated descriptors. First attribute value wins if there are duplicates.
```

Comment:
Add enough to this comment to make it obvious the `else` behaviour is intended. `RenderUnboundAttribute()` method name raises questions otherwise. For example ... "Treat later values of bound attributes as if they were unbound."